Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -158,6 +154,88 @@ public boolean onPreferenceChange(Preference preference, Object o) {
         });
     }
 
+    private static class SignOutAsyncTask extends AsyncTask<Void, Void, Boolean> {
+        private WeakReference<PreferencesFragment> fragmentWeakReference;
+
+        SignOutAsyncTask(PreferencesFragment fragment) {
+            fragmentWeakReference = new WeakReference<>(fragment);
+        }
+
+        @Override
+        protected Boolean doInBackground(Void... voids) {
+            PreferencesFragment fragment = fragmentWeakReference.get();
+            return fragment == null || fragment.hasUnsyncedNotes();
+        }
+
+        @Override
+        protected void onPostExecute(Boolean hasUnsyncedNotes) {
+            PreferencesFragment fragment = fragmentWeakReference.get();
+            if (fragment == null) {
+                return;
+            }
+
+            // Safety first! Check if any notes are unsynced and warn the user if so.
+            if (hasUnsyncedNotes) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(fragment.getContext());
```

Comment:
There maybe no need for this variable here. And this may be more readable since it's a `builder`:
```java
new AlertDialog.Builder(fragment.getContext())
        .setTitle(R.string.unsynced_notes)
        .setMessage(R.string.unsynced_notes_message)
        .setPositiveButton(R.string.delete_notes, fragment.signOutClickListener)
        .setNeutralButton(R.string.visit_web_app, fragment.loadWebAppClickListener)
        .setNegativeButton(R.string.cancel, null)
        .show();
```

or 

```java
AlertDialog.Builder builder = new AlertDialog.Builder(fragment.getContext())
        .setTitle(R.string.unsynced_notes)
        .setMessage(R.string.unsynced_notes_message)
        .setPositiveButton(R.string.delete_notes, fragment.signOutClickListener)
        .setNeutralButton(R.string.visit_web_app, fragment.loadWebAppClickListener)
        .setNegativeButton(R.string.cancel, null);
builder.show();
```